## Scott 10K Allied health dataset

### Aim: Obtain demographic, clinical and neuroimaging (T1 file paths, MWC1T1 file paths) for the scans

### Prelims: make scripts folder

In [1]:
%%bash

mkdir -p /rds/general/project/c3nl_scott_students/ephemeral/sankeith/scott_10k_scripts/alliedhealth

### First order of business: create the .txt file of all ADNI T1 patients I want in my table

In [2]:
%%bash 

mydir=/rds/general/project/scott_data_adni/live/ADNI/ADNI_NIFTI/
output_file=/rds/general/project/c3nl_scott_students/ephemeral/sankeith/scott_10k_lists/scott_10k_t1_paths.txt
>"$output_file"
find "$mydir" -type f -name "*.nii.*" >> "$output_file"

wc -l<"$output_file"

15733


In [3]:
import pandas as pd
import numpy as np
import os
from datetime import datetime, timedelta

t1_paths = '/rds/general/project/c3nl_scott_students/ephemeral/sankeith/scott_10k_lists/scott_10k_t1_paths.txt'
patient_id_list = []
site_id_list = []
rid_list = []
data_key_list = []
t1_year_list = []
t1_mon_list = []
t1_day_list = []
visdate_list = []

with open(t1_paths,'r') as paths:
    all_paths = paths.readlines()
    for path in all_paths:
        path_parts = path.split('/')
        patient = path_parts[-1].strip('.nii.gz\n')

        patient_parts = patient.split('_')
        data_key = '_'.join(patient_parts)
        data_key_list.append(data_key)
    
        patient_id = '_'.join(patient_parts[:3])
        rid = patient_id.split('_')[-1]
        site_id = patient_id.split('_')[0]
        site_id_list.append(site_id)
        rid_list.append(rid)
        patient_id_list.append(patient_id)
    
        try:
            t1_date = patient_parts[4]
            t1_year, t1_mon, t1_day = t1_date.split('-')
            t1_year_list.append(t1_year)
            t1_mon_list.append(t1_mon)
            t1_day_list.append(t1_day)
            visdate_list.append(f"{t1_year}-{t1_mon}-{t1_day}")
        except (IndexError, ValueError):
            t1_year_list.append(np.nan)
            t1_mon_list.append(np.nan)
            t1_day_list.append(np.nan)
            visdate_list.append(np.nan)
        
df = {
    "PTID" : patient_id_list,
    "RID" : rid_list,
    "SITE_ID" : site_id_list,
    "DATA_KEY" : data_key_list,
    "T1_YEAR" : t1_year_list,
    "T1_MON" : t1_mon_list,
    "T1_DAY" : t1_day_list,
    "VISDATE_STR" : visdate_list
}
df = pd.DataFrame(df)

df['T1_YEAR'] = pd.to_numeric(df['T1_YEAR'], errors='coerce')
df['T1_MON'] = pd.to_numeric(df['T1_MON'], errors='coerce')
df['T1_DAY'] = pd.to_numeric(df['T1_DAY'], errors='coerce')

df['VISDATE'] = pd.to_datetime(df['VISDATE_STR'], errors='coerce')
df = df.drop(columns=['VISDATE_STR'])

df['EXAMDATE_4WKS_LATER'] = df['VISDATE'] + timedelta(weeks = 4)
df['EXAMDATE_4WKS_B4'] = df['VISDATE'] - timedelta(weeks = 4)

df      

,PTID,RID,SITE_ID,DATA_KEY,T1_YEAR,T1_MON,T1_DAY,VISDATE,EXAMDATE_4WKS_LATER,EXAMDATE_4WKS_B4
0,012_S_4849,4849,012,012_S_4849_ADNI-T1_2012-11-05_12_04_44.0,2012.0,11.0,5.0,2012-11-05,2012-12-03,2012-10-08
1,012_S_4849,4849,012,012_S_4849_ADNI-T1_2012-07-24_09_50_58.0,2012.0,7.0,24.0,2012-07-24,2012-08-21,2012-06-26
2,012_S_4849,4849,012,012_S_4849_ADNI-T1_2013-08-26_14_02_58.0,2013.0,8.0,26.0,2013-08-26,2013-09-23,2013-07-29
3,012_S_4849,4849,012,012_S_4849_ADNI-T1_2013-02-25_13_24_41.0,2013.0,2.0,25.0,2013-02-25,2013-03-25,2013-01-28
4,941_S_6471,6471,941,941_S_6471_ADNI-T1_2020-09-01_12_46_08.0,2020.0,9.0,1.0,2020-09-01,2020-09-29,2020-08-04
...,...,...,...,...,...,...,...,...,...,...
15728,073_S_0312,0312,073,073_S_0312_ADNI-T1_2006-05-31_13_21_45.0,2006.0,5.0,31.0,2006-05-31,2006-06-28,2006-05-03
15729,073_S_0312,0312,073,073_S_0312_ADNI-T1_2007-02-08_16_25_18.0,2007.0,2.0,8.0,2007-02-08,2007-03-08,2007-01-11
15730,073_S_0312,0312,073,073_S_0312_ADNI-T1_2006-05-31_13_29_57.0,2006.0,5.0,31.0,2006-05-31,2006-06-28,2006-05-03
15731,073_S_0312,0312,073,073_S_0312_ADNI-T1_2007-02-08_16_33_35.0,2007.0,2.0,8.0,2007-02-08,2007-03-08,2007-01-11


### Time to add T1 paths

In [4]:
t1_paths = '/rds/general/project/c3nl_scott_students/ephemeral/sankeith/scott_10k_lists/scott_10k_t1_paths.txt'
key_to_path = {}

with open(t1_paths,'r') as paths:
    all_paths = paths.readlines()
    for path in all_paths:
        path  = path.strip('\n')
        raw_key = path.split('/')[-1]
        key = raw_key.replace('.nii.gz','')
        key_to_path[key] = path

df['T1_PATH'] = df['DATA_KEY'].map(key_to_path)
df

,PTID,RID,SITE_ID,DATA_KEY,T1_YEAR,T1_MON,T1_DAY,VISDATE,EXAMDATE_4WKS_LATER,EXAMDATE_4WKS_B4,T1_PATH
0,012_S_4849,4849,012,012_S_4849_ADNI-T1_2012-11-05_12_04_44.0,2012.0,11.0,5.0,2012-11-05,2012-12-03,2012-10-08,/rds/general/project/scott_data_adni/live/ADNI...
1,012_S_4849,4849,012,012_S_4849_ADNI-T1_2012-07-24_09_50_58.0,2012.0,7.0,24.0,2012-07-24,2012-08-21,2012-06-26,/rds/general/project/scott_data_adni/live/ADNI...
2,012_S_4849,4849,012,012_S_4849_ADNI-T1_2013-08-26_14_02_58.0,2013.0,8.0,26.0,2013-08-26,2013-09-23,2013-07-29,/rds/general/project/scott_data_adni/live/ADNI...
3,012_S_4849,4849,012,012_S_4849_ADNI-T1_2013-02-25_13_24_41.0,2013.0,2.0,25.0,2013-02-25,2013-03-25,2013-01-28,/rds/general/project/scott_data_adni/live/ADNI...
4,941_S_6471,6471,941,941_S_6471_ADNI-T1_2020-09-01_12_46_08.0,2020.0,9.0,1.0,2020-09-01,2020-09-29,2020-08-04,/rds/general/project/scott_data_adni/live/ADNI...
...,...,...,...,...,...,...,...,...,...,...,...
15728,073_S_0312,0312,073,073_S_0312_ADNI-T1_2006-05-31_13_21_45.0,2006.0,5.0,31.0,2006-05-31,2006-06-28,2006-05-03,/rds/general/project/scott_data_adni/live/ADNI...
15729,073_S_0312,0312,073,073_S_0312_ADNI-T1_2007-02-08_16_25_18.0,2007.0,2.0,8.0,2007-02-08,2007-03-08,2007-01-11,/rds/general/project/scott_data_adni/live/ADNI...
15730,073_S_0312,0312,073,073_S_0312_ADNI-T1_2006-05-31_13_29_57.0,2006.0,5.0,31.0,2006-05-31,2006-06-28,2006-05-03,/rds/general/project/scott_data_adni/live/ADNI...
15731,073_S_0312,0312,073,073_S_0312_ADNI-T1_2007-02-08_16_33_35.0,2007.0,2.0,8.0,2007-02-08,2007-03-08,2007-01-11,/rds/general/project/scott_data_adni/live/ADNI...


### Now for mwc1t1 paths

In [5]:
%%bash

mydir=/rds/general/project/c3nl_scott_students/ephemeral/sankeith/scott_10k_20aug25/
output_file=/rds/general/project/c3nl_scott_students/ephemeral/sankeith/scott_10k_lists/successful_scott_10k_mwc1t1_paths.txt

### collect mwc1t1 file paths##

>"$output_file"
find "$mydir" -type f -name "mwc1t1*" >> "$output_file"
wc -l<"${output_file}"

15726


In [6]:
mwc1t1_paths = '/rds/general/project/c3nl_scott_students/ephemeral/sankeith/scott_10k_lists/successful_scott_10k_mwc1t1_paths.txt'
key_to_path = {}

with open(mwc1t1_paths,'r') as paths:
    all_paths = paths.readlines()
    for path in all_paths:
        path  = path.strip('\n')
        key = path.split('/')[-2]
        key_to_path[key] = path

df['MWC1T1_PATH'] = df['DATA_KEY'].map(key_to_path)
df

,PTID,RID,SITE_ID,DATA_KEY,T1_YEAR,T1_MON,T1_DAY,VISDATE,EXAMDATE_4WKS_LATER,EXAMDATE_4WKS_B4,T1_PATH,MWC1T1_PATH
0,012_S_4849,4849,012,012_S_4849_ADNI-T1_2012-11-05_12_04_44.0,2012.0,11.0,5.0,2012-11-05,2012-12-03,2012-10-08,/rds/general/project/scott_data_adni/live/ADNI...,/rds/general/project/c3nl_scott_students/ephem...
1,012_S_4849,4849,012,012_S_4849_ADNI-T1_2012-07-24_09_50_58.0,2012.0,7.0,24.0,2012-07-24,2012-08-21,2012-06-26,/rds/general/project/scott_data_adni/live/ADNI...,/rds/general/project/c3nl_scott_students/ephem...
2,012_S_4849,4849,012,012_S_4849_ADNI-T1_2013-08-26_14_02_58.0,2013.0,8.0,26.0,2013-08-26,2013-09-23,2013-07-29,/rds/general/project/scott_data_adni/live/ADNI...,/rds/general/project/c3nl_scott_students/ephem...
3,012_S_4849,4849,012,012_S_4849_ADNI-T1_2013-02-25_13_24_41.0,2013.0,2.0,25.0,2013-02-25,2013-03-25,2013-01-28,/rds/general/project/scott_data_adni/live/ADNI...,/rds/general/project/c3nl_scott_students/ephem...
4,941_S_6471,6471,941,941_S_6471_ADNI-T1_2020-09-01_12_46_08.0,2020.0,9.0,1.0,2020-09-01,2020-09-29,2020-08-04,/rds/general/project/scott_data_adni/live/ADNI...,/rds/general/project/c3nl_scott_students/ephem...
...,...,...,...,...,...,...,...,...,...,...,...,...
15728,073_S_0312,0312,073,073_S_0312_ADNI-T1_2006-05-31_13_21_45.0,2006.0,5.0,31.0,2006-05-31,2006-06-28,2006-05-03,/rds/general/project/scott_data_adni/live/ADNI...,/rds/general/project/c3nl_scott_students/ephem...
15729,073_S_0312,0312,073,073_S_0312_ADNI-T1_2007-02-08_16_25_18.0,2007.0,2.0,8.0,2007-02-08,2007-03-08,2007-01-11,/rds/general/project/scott_data_adni/live/ADNI...,/rds/general/project/c3nl_scott_students/ephem...
15730,073_S_0312,0312,073,073_S_0312_ADNI-T1_2006-05-31_13_29_57.0,2006.0,5.0,31.0,2006-05-31,2006-06-28,2006-05-03,/rds/general/project/scott_data_adni/live/ADNI...,/rds/general/project/c3nl_scott_students/ephem...
15731,073_S_0312,0312,073,073_S_0312_ADNI-T1_2007-02-08_16_33_35.0,2007.0,2.0,8.0,2007-02-08,2007-03-08,2007-01-11,/rds/general/project/scott_data_adni/live/ADNI...,/rds/general/project/c3nl_scott_students/ephem...


#### How many unique patients are there (as measured by PTID)

In [6]:
len(df['PTID'].unique())

2635

In [ ]:
df.to_csv('/rds/general/project/c3nl_scott_students/ephemeral/sankeith/scott_10k_scripts/alliedhealth/interim_scott_10k_alliedhealth.csv', index = False)

### Make a .m file of Greg's code, and run that. This gets us patient demographics (Gender, Date of Birth and Education)

In [2]:
%%file /rds/general/project/c3nl_scott_students/ephemeral/sankeith/scott_10k_scripts/alliedhealth/scott_10k_add_demographics_info.m

% Load base table
T = readtable('/rds/general/project/c3nl_scott_students/ephemeral/sankeith/scott_10k_scripts/alliedhealth/interim_scott_10k_alliedhealth.csv');

% Save .mat copy
save('/rds/general/project/c3nl_scott_students/ephemeral/sankeith/scott_10k_scripts/alliedhealth/interim_scott_10k_alliedhealth.mat');

data = load('/rds/general/project/c3nl_scott_students/ephemeral/sankeith/scott_10k_scripts/alliedhealth/interim_scott_10k_alliedhealth.mat');
disp(data)

fprintf('Base table loaded: Width=%d Height=%d\n', width(T), height(T));

%% Define auxiliary files and variables
vars = { ...
{ ...
'/rds/general/project/c3nl_scott_students/live/sankeith/scott_10k_housekeeping/scott_10k_extra_adni_info/PTDEMOG_10Sep2025.csv', ...
{'PTGENDER', 'PTDOB', 'PTEDUCAT'}, ...
{NaN, NaT, NaN}, ...
{[], @(x) datetime(x,'InputFormat','MM/yyyy'), []} ...
} ...
};

%% Main joining loop
for v = 1:numel(vars)

fprintf('\n------------------------------------------\n');
fprintf('Processing file: %s\n', vars{v}{1});

% Load auxiliary table
Talli = readtable(vars{v}{1});

% Ensure PTID is string for matching
Talli.PTID = string(Talli.PTID);

% Ensure VISDATE exists
if ~ismember('VISDATE', Talli.Properties.VariableNames)
    Talli.VISDATE = Talli.EXAMDATE;
end

fprintf('Auxiliary rows: %d\n', height(Talli));

varNames = vars{v}{2};

%% Initialise columns in T with missing values
fprintf('(initialising columns)\n');

if ~isempty(vars{v}{3})

    for vf = 1:numel(vars{v}{3})

        default_value = vars{v}{3}{vf};

        T.(varNames{vf}) = repmat(default_value, height(T), 1);

    end

else

    T{:,varNames} = nan(height(T), numel(varNames));

end

%% Convert date formats if required
fprintf('(converting date formats)\n');

if ~isempty(vars{v}{4})

    for vf = 1:numel(vars{v}{4})

        converter = vars{v}{4}{vf};

        if ~isempty(converter)

            col = varNames{vf};

            Talli.(col) = converter(Talli.(col));

        end

    end

end

fprintf('(date conversion done)\n');

%% Clean ADNI sentinel values in source table (BEST PRACTICE LOCATION)
fprintf('(cleaning sentinel values in auxiliary table)\n');

for vf = 1:numel(varNames)

    col = varNames{vf};

    if isnumeric(Talli.(col))

        badmask = (Talli.(col) == -4.0 | Talli.(col) == -1.0);

        Talli.(col)(badmask) = NaN;

    end

end

%% Match each subject row
fprintf('(matching rows by PTID and VISDATE)\n');

for s = 1:height(T)

    % Find matching subject
    Tsub = Talli(Talli.PTID == T.PTID(s), :);

    if ~isempty(Tsub)

        % Compute date differences
        Tsub.datediffs = abs(days(Tsub.VISDATE - T.VISDATE(s)));

        % Sort so nearest date is first
        Tsub = sortrows(Tsub, 'datediffs', 'asc');

        % Assign closest row values
        T(s, varNames) = Tsub(1, varNames);

    end

end

fprintf('(finished processing this file)\n');

end

%% Final safety cleaning (optional redundancy)
fprintf('\nFinal sentinel cleaning pass on output table\n');

for v = 1:numel(vars)

varNames = vars{v}{2};

for vf = 1:numel(varNames)

    col = varNames{vf};

    if isnumeric(T.(col))

        badmask = (T.(col) == -4.0 | T.(col) == -1.0);

        T.(col)(badmask) = NaN;

    end

end

end

%% Save result
output_file = '/rds/general/project/c3nl_scott_students/ephemeral/sankeith/scott_10k_scripts/alliedhealth/gregbased_scott10k_alliedhealth.csv';

writetable(T, output_file);

fprintf('\n------------------------------------------\n');
fprintf('Final table saved\n');
fprintf('Width: %d Height: %d\n', width(T), height(T));

if isfile(output_file)
disp('Demographics data joined successfully')
else
warning('Output file was not created')
end


Writing /rds/general/project/c3nl_scott_students/ephemeral/sankeith/scott_10k_scripts/alliedhealth/scott_10k_add_demographics_info.m


In [3]:
%%file /rds/general/project/c3nl_scott_students/ephemeral/sankeith/scott_10k_scripts/alliedhealth/scott_10k_add_demographics_info.sh

#!/bin/bash

module load tools/prod
module --ignore_cache load MATLAB2024/b > /dev/null 2>&1

matlab -nosplash -nodesktop -r "run('/rds/general/project/c3nl_scott_students/ephemeral/sankeith/scott_10k_scripts/alliedhealth/scott_10k_add_demographics_info.m'); exit"

rm -f ~/java* ~/*crash*dump*

Writing /rds/general/project/c3nl_scott_students/ephemeral/sankeith/scott_10k_scripts/alliedhealth/scott_10k_add_demographics_info.sh


In [ ]:
%%bash

chmod -Rf 775 /rds/general/project/c3nl_scott_students/ephemeral/sankeith/scott_10k_scripts/alliedhealth/scott_10k_add_demographics_info.*

/rds/general/project/c3nl_scott_students/ephemeral/sankeith/scott_10k_scripts/alliedhealth/scott_10k_add_demographics_info.sh

### Load in time-sensitive data

In [11]:
import pandas as pd
import os
import numpy as np
import datetime
from datetime import timedelta
from dateutil.relativedelta import relativedelta
import fnmatch
import warnings
import math
warnings.simplefilter(action='ignore', category=FutureWarning)

In [ ]:
df = pd.read_csv('/rds/general/project/c3nl_scott_students/ephemeral/sankeith/scott_10k_scripts/alliedhealth/gregbased_scott10k_alliedhealth.csv', low_memory = False)
df[f'VISDATE'] = pd.to_datetime(df[f'VISDATE'])
df[f'EXAMDATE_4WKS_LATER'] = pd.to_datetime(df[f'EXAMDATE_4WKS_LATER'])
df[f'EXAMDATE_4WKS_B4'] = pd.to_datetime(df[f'EXAMDATE_4WKS_B4'])
df[f'PTDOB'] = pd.to_datetime(df[f'PTDOB'])
df[f'PTAGE_YEARS'] = df['VISDATE'] - df['PTDOB']
df[f'PTAGE'] = df[f'PTAGE_YEARS'] / timedelta(days = 365)
def floor_years(n):
    try:
        return math.floor(n)
    except ValueError:
        return n
df[f'PTAGE'] = df[f'PTAGE'].apply(lambda x: floor_years(x))
df.drop(columns = 'PTAGE_YEARS', inplace = True)
print(len(df))
print(len(df.columns))
print(len(df['PTID'].unique()))
print(df[f'PTAGE'].describe())
print(df['PTAGE'].value_counts()) ## To check there aren't any weird -4.0 or -1.0 values

### Merge APOE Genotype

In [13]:
apoe_df = pd.read_csv('/rds/general/project/c3nl_scott_students/live/sankeith/scott_10k_housekeeping/scott_10k_extra_adni_info/APOERES_28Mar2025.csv',low_memory = False)[['PTID','GENOTYPE']]
apoe_df.drop_duplicates(subset = ['PTID','GENOTYPE'], inplace = True)
print(apoe_df)
df = df.merge(apoe_df, how = 'left', on = ['PTID'])
print(len(df))
print(df['GENOTYPE'].value_counts()) ## To check there aren't any weird -4.0 or -1.0 values

            PTID GENOTYPE
0     011_S_0002      3/3
1     011_S_0003      3/4
2     022_S_0004      3/3
3     011_S_0005      3/3
4     022_S_0007      3/4
...          ...      ...
2755  341_S_7018      3/3
2756  035_S_7019      3/3
2757  052_S_7027      4/4
2758  941_S_7041      3/4
2759  035_S_7049      3/3

[2760 rows x 2 columns]
15733
GENOTYPE
3/3    7327
3/4    5130
4/4    1522
2/3    1105
2/4     313
2/2      30
Name: count, dtype: int64


### Prepping ADAS13 DataFrame and MRI FIELD STRENGTH DataFrame

In [14]:
### Prepping ADAS123 DataFrame ###
adas_adni1_df = pd.read_csv('/rds/general/project/c3nl_scott_students/live/sankeith/scott_10k_housekeeping/scott_10k_extra_adni_info/ADASSCORES_01Apr2025.csv', low_memory = False)[['PTID','EXAMDATE','TOTALMOD']]
adas_adni1_df.rename(columns={'TOTALMOD': 'TOTAL13', 'EXAMDATE': 'VISDATE'}, inplace=True)

adas_adnigo23_df = pd.read_csv('/rds/general/project/c3nl_scott_students/live/sankeith/scott_10k_housekeeping/scott_10k_extra_adni_info/ADAS_ADNIGO23_31Mar2025.csv', low_memory = False)
adas_adnigo23_df = adas_adnigo23_df[['PTID','VISDATE','TOTAL13']]

adas_df = pd.concat([adas_adni1_df, adas_adnigo23_df], ignore_index=True)

print(adas_df['TOTAL13'].value_counts()) ## To check there aren't any weird -4.0 or -1.0 values
print(len(adas_df))

TOTAL13
9.00     392
7.00     374
11.00    372
8.00     346
10.00    344
        ... 
77.00      1
74.00      1
68.33      1
63.67      1
67.33      1
Name: count, Length: 208, dtype: int64
11618


In [15]:
adas_df.to_csv(f'/rds/general/project/c3nl_scott_students/live/sankeith/scott_10k_housekeeping/scott_10k_extra_adni_info/ADAS13_{datetime.datetime.now().strftime('%d%b%Y')}.csv', index = False)

### Prepping MRI FIELD STRENGTH DataFrame ###

mrimeta_df = pd.read_csv('/rds/general/project/c3nl_scott_students/live/sankeith/scott_10k_housekeeping/scott_10k_extra_adni_info/MRIMETA_11Sep2025.csv', low_memory = False)[['PTID','EXAMDATE','FIELD_STRENGTH']]
mrimeta_3t_df = pd.read_csv('/rds/general/project/c3nl_scott_students/live/sankeith/scott_10k_housekeeping/scott_10k_extra_adni_info/MRI3META_11Sep2025.csv', low_memory = False)[['PTID','EXAMDATE','FIELD_STRENGTH']]
mri_df = pd.concat([mrimeta_df, mrimeta_3t_df], ignore_index=True)
print(len(mri_df))
print(mri_df['FIELD_STRENGTH'].value_counts()) ## To check there aren't any weird -4.0 or -1.0 values
mri_df.to_csv(f'/rds/general/project/c3nl_scott_students/live/sankeith/scott_10k_housekeeping/scott_10k_extra_adni_info/MRIFLDSTRNGTH_{datetime.datetime.now().strftime('%d%b%Y')}.csv', index = False)

13933
FIELD_STRENGTH
3T      9193
1.5T    4740
Name: count, dtype: int64


In [16]:
### Prepping Amyloid DataFrame - I only use the batemanlab info which includes EXAMDATE (the other batemanlab DataFrame only has VISCODE, so can't join with Scott 10K) ###

amyloid_df = pd.read_csv('/rds/general/project/c3nl_scott_students/live/sankeith/scott_10k_housekeeping/scott_10k_extra_adni_info/batemanlab_20221118_31Mar2025.csv', low_memory = False)[['RID' ,'EXAMDATE', 'Abeta_4240_Standardized', 'Intertcept_Standardization', 'Slope_Standardization', 'Sample_volume', 'Sample_volume_UNITS', 'Abeta_42_conc', 'Abeta_42_conc_UNITS', 'Abeta_42_N14N15', 'Abeta_42_N15_ISTD_amount', 'Abeta_42_N15_ISTD_amount_UNITS', 'Abeta_40_conc', 'Abeta_40_conc_UNITS', 'Abeta_40_N14N15', 'Abeta_40_N15_ISTD_amount', 'Abeta_40_N15_ISTD_amount_UNITS', 'Abeta_4240']]
print(len(amyloid_df))
for col in amyloid_df.columns.tolist():
    print(amyloid_df[col].value_counts())
amyloid_df.to_csv(f'/rds/general/project/c3nl_scott_students/live/sankeith/scott_10k_housekeeping/scott_10k_extra_adni_info/AMYLOID_{datetime.datetime.now().strftime('%d%b%Y')}.csv', index = False)

742
RID
1169.0    8
260.0     7
746.0     6
56.0      6
618.0     6
         ..
4715.0    1
214.0     1
126.0     1
4757.0    1
4254.0    1
Name: count, Length: 358, dtype: int64
EXAMDATE
2012-02-02    4
2012-02-22    3
2010-12-15    3
2012-02-23    3
2011-07-11    3
             ..
2009-06-30    1
2007-06-26    1
2010-07-02    1
2018-03-14    1
2012-01-06    1
Name: count, Length: 505, dtype: int64
Abeta_4240_Standardized
0.118630    1
0.132158    1
0.103956    1
0.130436    1
0.131871    1
           ..
0.128109    1
0.142453    1
0.120423    1
0.103758    1
0.128779    1
Name: count, Length: 728, dtype: int64
Intertcept_Standardization
 0.032290    98
 0.000965    98
-0.011867    97
-0.032107    97
-0.010690    96
-0.021845    94
-0.000960    90
 0.017235    58
Name: count, dtype: int64
Slope_Standardization
0.724008    98
0.997231    98
1.092096    97
1.262250    97
1.075215    96
1.194029    94
1.015438    90
0.865222    58
Name: count, dtype: int64
Sample_volume
0.45    727
0.43 

In [17]:
### Prepping p217 tau DataFrame ###

tau_df = pd.read_csv('/rds/general/project/c3nl_scott_students/live/sankeith/scott_10k_housekeeping/scott_10k_extra_adni_info/JANSSEN_PLASMA_P217_TAU_01Apr2025.csv', low_memory = False)[['PTID', 'EXAMDATE', 'DILUTION_CORRECTED_CONC', 'CV']]
tau_df.rename(columns={'DILUTION_CORRECTED_CONC': 'P217_DILUTION_CORRECTED_CONC', 'CV': 'P217_CV'}, inplace=True)
tau_df.to_csv(f'/rds/general/project/c3nl_scott_students/live/sankeith/scott_10k_housekeeping/scott_10k_extra_adni_info/P217_{datetime.datetime.now().strftime('%d%b%Y')}.csv', index = False)

### Merging time-sensitive visit information (based on Greg's code):
1. Load an allied health file
2. Ensure allied health file as examdate
3. Do a right type join (assume left = og file, right = allied health file)
4. Extract rows only with a T1 file path, so we have full rows
5. Calculate abs(allied health data - MRI visit date) and label it like df.datediffs or some shit
6. df.groupby with PTID, and then sort by datediffs ascend
7. Keep the row where datadiff is less than 28 days and the smallest val. If tie, then we ball and just choose at random
8. Conduct a final length of rows check - MUST BE 15733
9. Rinse and repeat

In [18]:
vars = [
    [f'/rds/general/project/c3nl_scott_students/live/sankeith/scott_10k_housekeeping/scott_10k_extra_adni_info/MRIFLDSTRNGTH_{datetime.datetime.now().strftime('%d%b%Y')}.csv',['PTID','EXAMDATE','FIELD_STRENGTH'], {'FIELD_STRENGTH': pd.NA}],
    ['/rds/general/project/c3nl_scott_students/live/sankeith/scott_10k_housekeeping/scott_10k_extra_adni_info/DXSUM_10Sep2025.csv',['PTID','EXAMDATE','PHASE', 'DIAGNOSIS'], {'PHASE': pd.NA,'DIAGNOSIS':np.nan}],
    ['/rds/general/project/c3nl_scott_students/live/sankeith/scott_10k_housekeeping/scott_10k_extra_adni_info/MMSE_28Mar2025.csv', ['PTID','VISDATE','MMSCORE'], {'MMSCORE':np.nan}],
    ['/rds/general/project/c3nl_scott_students/live/sankeith/scott_10k_housekeeping/scott_10k_extra_adni_info/MOCA_30Mar2025.csv', ['PTID','VISDATE', 'MOCA'], {'MOCA':np.nan}],
    ['/rds/general/project/c3nl_scott_students/live/sankeith/scott_10k_housekeeping/scott_10k_extra_adni_info/NEUROBAT_28Mar2025.csv', ['PTID','VISDATE','LIMMTOTAL', 'CLOCKSCOR', 'LDELTOTAL', 'LDELCUE', 'ANART'], {'LIMMTOTAL':np.nan, 'CLOCKSCOR':np.nan, 'LDELTOTAL':np.nan, 'LDELCUE':np.nan, 'ANART':np.nan}],
    ['/rds/general/project/c3nl_scott_students/live/sankeith/scott_10k_housekeeping/scott_10k_extra_adni_info/CDR_28Mar2025.csv', ['PTID','VISDATE', 'CDGLOBAL'], {'CDGLOBAL':np.nan}],
    ['/rds/general/project/c3nl_scott_students/live/sankeith/scott_10k_housekeeping/scott_10k_extra_adni_info/FAQ_28Mar2025.csv', ['PTID','VISDATE', 'FAQTOTAL'], {'FAQTOTAL':np.nan}],
    ['/rds/general/project/c3nl_scott_students/live/sankeith/scott_10k_housekeeping/scott_10k_extra_adni_info/NPIQ_28Mar2025.csv',['PTID', 'VISDATE','NPISCORE'], {'NPISCORE':np.nan}],
    ['/rds/general/project/c3nl_scott_students/live/sankeith/scott_10k_housekeeping/scott_10k_extra_adni_info/GDSCALE_28Mar2025.csv',['PTID','VISDATE','GDTOTAL'], {'GDTOTAL':np.nan}],
    ['/rds/general/project/c3nl_scott_students/live/sankeith/scott_10k_housekeeping/scott_10k_extra_adni_info/MODHACH_01Apr2025.csv',['PTID', 'VISDATE','HMSCORE'], {'HMSCORE':np.nan}],
    [f'/rds/general/project/c3nl_scott_students/live/sankeith/scott_10k_housekeeping/scott_10k_extra_adni_info/ADAS13_{datetime.datetime.now().strftime('%d%b%Y')}.csv',['PTID','VISDATE','TOTAL13'], {'TOTAL13':np.nan}],
    [f'/rds/general/project/c3nl_scott_students/live/sankeith/scott_10k_housekeeping/scott_10k_extra_adni_info/P217_{datetime.datetime.now().strftime('%d%b%Y')}.csv',['PTID', 'EXAMDATE', 'P217_DILUTION_CORRECTED_CONC', 'P217_CV'], {'P217_DILUTION_CORRECTED_CONC': np.nan, 'P217_CV': np.nan}]
]
for i in range(0,len(vars)):
    column_suffix = vars[i][0].split('/')[-1]
    column_suffix = column_suffix.split('_')[0]
    varnames = vars[i][1]
    
    #Load an allied health file
    Talli = pd.read_csv(vars[i][0], low_memory = False)
    Talli = Talli[vars[i][1]]
    
    if 'RID' in Talli.columns:
        Talli.drop(columns = 'RID', inplace = True)    
    print(f"Length of {column_suffix}: {len(Talli)}")
    
    #Ensure allied health file as examdate
    if 'VISDATE' in Talli.columns:
        Talli.rename(columns = {'VISDATE':f"EXAMDATE_{column_suffix}"}, inplace = True)
        Talli[f'EXAMDATE_{column_suffix}'] = pd.to_datetime(Talli[f'EXAMDATE_{column_suffix}'], errors = 'coerce')
    else:
        Talli[f'EXAMDATE'] = pd.to_datetime(Talli[f'EXAMDATE'], errors = 'coerce')
        Talli.rename(columns = {'EXAMDATE': f'EXAMDATE_{column_suffix}'}, inplace = True)

    #Do a left type join
    Tsub = pd.merge(df,Talli, on = 'PTID', how = 'left')
    print(f"Length of the freshly joined dataframe: {len(Tsub)}")

    #Extract rows only with a T1 file path, so we have full rows
    Tsub.dropna(subset='T1_PATH',inplace=True)
    print(f"After dropping rows without T1 paths: {len(Tsub)}")

    #Calculate abs(allied health data - MRI visit date) and label it like df.datediffs or some shit
    try:
        Tsub[f'DATEDIFFS_{column_suffix}'] = abs(Tsub[f'EXAMDATE_{column_suffix}'] - Tsub['VISDATE'])
    except KeyError:
        Tsub[f'DATEDIFFS_{column_suffix}'] = abs(Tsub['EXAMDATE'] - Tsub['VISDATE'])

    #df.groupby with PTID, and then sort by datediffs ascend. Select the/a row where DATEDIFFS is the smallest
    Tsub = Tsub.sort_values(f'DATEDIFFS_{column_suffix}', ascending=True)

    print(f"Grouping rows by 'T1_PATH'. Within groups, organising rows so smallest DATE_DIFFS_{column_suffix} are at the top") 
    min_diffs = (
    Tsub
    .groupby('T1_PATH')[f'DATEDIFFS_{column_suffix}']
    .transform('min')
    )
    
    Tsub = Tsub[
    (Tsub[f'DATEDIFFS_{column_suffix}'] == min_diffs) |
    (min_diffs.isna() & Tsub[f'DATEDIFFS_{column_suffix}'].isna())
    ]

    Tsub.reset_index(drop=True, inplace=True)

    #Filter logic: if DATE_DIFFS > 28 days or no EXAMDATE, then either the data is too old to be used or we can't verify if it's too old to be used. Thus, we need to replace the relevant values with blanks
    print(f"Finding rows where *EITHER* DATE_DIFFS_{column_suffix} > 28 days *OR* there is no EXAMDATE_{column_suffix} column. If that's the case, replace the relevant values with NaNs, NaTs and pd.NAs") 
    mask = (Tsub[f'DATEDIFFS_{column_suffix}'] > pd.Timedelta(28, "D")) | (Tsub[f'EXAMDATE_{column_suffix}'].isna() == True)
    Tsub.loc[mask, f'DATEDIFFS_{column_suffix}'] = pd.NaT
    Tsub.loc[mask, f'EXAMDATE_{column_suffix}'] = pd.NaT
    for key in vars[i][2].keys():
        Tsub.loc[mask, key] = vars[i][2].get(key)

    #Filter logic: ADNI likes to use -4.0 or -1.0 for missing patient data. Make sure those are cleaned out and those values are replaced with NaNs
    for col in vars[i][2].keys():
        if col in Tsub.columns:
            Tsub.loc[Tsub[col].isin([-4.0, -1.0]), col] = np.nan


    print(f'Length of database before dropping exactly identical rows = {len(Tsub)}. Tsub is of type {type(Tsub)}')
    Tsub.drop_duplicates(keep = 'first',inplace = True)
    print(f'Length of database after dropping rows which are exactly identical = {len(Tsub)}.')
    
    dupe_mask = Tsub.duplicated(subset = 'T1_PATH',keep=False)
    Tsub_dupes = Tsub.loc[dupe_mask].copy()
    print(f'There are {len(Tsub_dupes)} rows with identical T1 MRI file paths, from {Tsub_dupes['T1_PATH'].nunique()} different MRI scans')

    def fewest_missing_one(group):
        missing_counts = group.isna().sum(axis=1)
        min_missing = missing_counts.min()
        candidates = group[missing_counts == min_missing]
        # if more than one, pick the first (or the row with smallest DATEDIFF)
        return candidates.iloc[[0]]  # keeps only 1 row

    print("Grouping rows by T1_PATH, finding the rows with the fewest missing values. If it's a tie between two rows, the first one for each T1_PATH group is kept")
    Tsub = Tsub.groupby('T1_PATH', group_keys=False).apply(fewest_missing_one, include_groups = True)
    Tsub.reset_index(drop=True, inplace=True)

    # Final sanity check
    dup_cols = [c for c in ['T1_PATH', 'MWC1T1_PATH'] if c in Tsub.columns]
    print(f"Final row count: {len(Tsub)}")
    print(f"Remaining duplicates: {Tsub.duplicated(subset=dup_cols).sum()}")
    
    # For each T1_PATH, see how many rows were kept
    rows_per_scan = Tsub.groupby('T1_PATH').size()
    print(rows_per_scan.value_counts())

    #Make the DATEDIFFS column numeric? Just because as datetime it says 'days' at the end and I don't like that.
    Tsub[f'DATEDIFFS_{column_suffix}'] = (Tsub[f'DATEDIFFS_{column_suffix}'].dt.days)
    
    print(Tsub[f'DATEDIFFS_{column_suffix}'].describe())
    print('')
    Tsub = Tsub.sort_values('T1_PATH').reset_index(drop=True)

    Tsub.to_csv('/rds/general/project/c3nl_scott_students/ephemeral/sankeith/scott_10k_scripts/alliedhealth/final_gregbased_scott10k_alliedhealth.csv', index = False)
    df = Tsub

Tsub.to_csv('/rds/general/project/c3nl_scott_students/ephemeral/sankeith/scott_10k_scripts/alliedhealth/scott10k_alliedhealth.csv', index=False)


Length of MRIFLDSTRNGTH: 13933
Length of the freshly joined dataframe: 118218
After dropping rows without T1 paths: 118218
Grouping rows by 'T1_PATH'. Within groups, organising rows so smallest DATE_DIFFS_MRIFLDSTRNGTH are at the top
Finding rows where *EITHER* DATE_DIFFS_MRIFLDSTRNGTH > 28 days *OR* there is no EXAMDATE_MRIFLDSTRNGTH column. If that's the case, replace the relevant values with NaNs, NaTs and pd.NAs
Length of database before dropping exactly identical rows = 16804. Tsub is of type <class 'pandas.core.frame.DataFrame'>
Length of database after dropping rows which are exactly identical = 16796.
There are 2126 rows with identical T1 MRI file paths, from 1063 different MRI scans
Grouping rows by T1_PATH, finding the rows with the fewest missing values. If it's a tie between two rows, the first one for each T1_PATH group is kept
Final row count: 15733
Remaining duplicates: 0
1    15733
Name: count, dtype: int64
count    15640.000000
mean         0.037148
std          0.6974

### Add Amyloid data

In [19]:
import pandas as pd
import numpy as np
import datetime
from datetime import timedelta

df = pd.read_csv('/rds/general/project/c3nl_scott_students/ephemeral/sankeith/scott_10k_scripts/alliedhealth/scott10k_alliedhealth.csv', low_memory = True)
df['VISDATE'] = pd.to_datetime(df['VISDATE'], errors='coerce')


vars = [
    [f'/rds/general/project/c3nl_scott_students/live/sankeith/scott_10k_housekeeping/scott_10k_extra_adni_info/AMYLOID_{datetime.datetime.now().strftime('%d%b%Y')}.csv',
     ['RID' ,'EXAMDATE', 'Abeta_4240_Standardized', 'Intertcept_Standardization', 'Slope_Standardization', 'Sample_volume', 'Sample_volume_UNITS', 'Abeta_42_conc', 'Abeta_42_conc_UNITS', 'Abeta_42_N14N15', 'Abeta_42_N15_ISTD_amount', 'Abeta_42_N15_ISTD_amount_UNITS', 'Abeta_40_conc', 'Abeta_40_conc_UNITS', 'Abeta_40_N14N15', 'Abeta_40_N15_ISTD_amount', 'Abeta_40_N15_ISTD_amount_UNITS', 'Abeta_4240'], 
     {'Abeta_4240_Standardized': np.nan, 
     'Intertcept_Standardization': np.nan,
     'Slope_Standardization': np.nan,
     'Sample_volume': np.nan,
     'Sample_volume_UNITS': pd.NA,
     'Abeta_42_conc': np.nan,
     'Abeta_42_conc_UNITS': pd.NA,
     'Abeta_42_N14N15': np.nan,
     'Abeta_42_N15_ISTD_amount': np.nan,
     'Abeta_42_N15_ISTD_amount_UNITS': pd.NA,
     'Abeta_40_conc': np.nan,
     'Abeta_40_conc_UNITS': pd.NA,
     'Abeta_40_N14N15': np.nan,
     'Abeta_40_N15_ISTD_amount': np.nan,
     'Abeta_40_N15_ISTD_amount_UNITS': pd.NA,
     'Abeta_4240': np.nan}]
]
for i in range(0,len(vars)):
    column_suffix = vars[i][0].split('/')[-1]
    column_suffix = column_suffix.split('_')[0]
    varnames = vars[i][1]
    
    #Load an allied health file
    Talli = pd.read_csv(vars[i][0], low_memory = False)
    Talli = Talli[vars[i][1]]
    
    #Ensure allied health file as examdate
    Talli[f'EXAMDATE'] = pd.to_datetime(Talli[f'EXAMDATE'], errors = 'coerce')
    Talli.rename(columns = {'EXAMDATE': f'EXAMDATE_{column_suffix}'}, inplace = True)

    #Do a left type join
    Tsub = pd.merge(df, Talli, on = 'RID', how = 'left')
    print(f"Length of the freshly joined dataframe: {len(Tsub)}")
    
    #Extract rows only with a T1 file path, so we have full rows
    Tsub.dropna(subset='T1_PATH',inplace=True)
    print(f"After dropping rows without T1 paths: {len(Tsub)}")

    #Calculate abs(allied health data - MRI visit date) and label it like df.datediffs or some shit
    Tsub[f'DATEDIFFS_{column_suffix}'] = abs(Tsub[f'EXAMDATE_{column_suffix}'] - Tsub['VISDATE'])

    #df.groupby with PTID, and then sort by datediffs ascend. Select the/a row where DATEDIFFS is the smallest
    Tsub = Tsub.sort_values(f'DATEDIFFS_{column_suffix}', ascending=True)

    print(f"Grouping rows by 'T1_PATH'. Within groups, organising rows so smallest DATE_DIFFS_{column_suffix} are at the top") 
    min_diffs = (
    Tsub
    .groupby('T1_PATH')[f'DATEDIFFS_{column_suffix}']
    .transform('min')
    )
    
    Tsub = Tsub[
    (Tsub[f'DATEDIFFS_{column_suffix}'] == min_diffs) |
    (min_diffs.isna() & Tsub[f'DATEDIFFS_{column_suffix}'].isna())
    ]

    Tsub.reset_index(drop=True, inplace=True)

    #Filter logic: if DATE_DIFFS > 28 days or no EXAMDATE, then either the data is too old to be used or we can't verify if it's too old to be used. Thus, we need to replace the relevant values with blanks
    print(f"Finding rows where *EITHER* DATE_DIFFS_{column_suffix} > 28 days *OR* there is no EXAMDATE_{column_suffix} column. If that's the case, replace the relevant values with NaNs, NaTs and pd.NAs") 
    mask = (Tsub[f'DATEDIFFS_{column_suffix}'] > pd.Timedelta(28, "D")) | (Tsub[f'EXAMDATE_{column_suffix}'].isna() == True)
    Tsub.loc[mask, f'DATEDIFFS_{column_suffix}'] = pd.NaT
    Tsub.loc[mask, f'EXAMDATE_{column_suffix}'] = pd.NaT
    for key in vars[i][2].keys():
        Tsub.loc[mask, key] = vars[i][2].get(key)

    #Filter logic: ADNI likes to use -4.0 or -1.0 for missing patient data. Make sure those are cleaned out and those values are replaced with NaNs
    for col in vars[i][2].keys():
        if col in Tsub.columns:
            Tsub.loc[Tsub[col].isin([-4.0, -1.0]), col] = np.nan

    print(f'Length of database before dropping exactly identical rows = {len(Tsub)}. Tsub is of type {type(Tsub)}')
    Tsub.drop_duplicates(keep = 'first',inplace = True)
    print(f'Length of database after dropping rows which are exactly identical = {len(Tsub)}.')
    
    dupe_mask = Tsub.duplicated(subset = 'T1_PATH',keep=False)
    Tsub_dupes = Tsub.loc[dupe_mask].copy()
    print(f'There are {len(Tsub_dupes)} rows with identical T1 MRI file paths, from {Tsub_dupes['T1_PATH'].nunique()} different MRI scans')

    def fewest_missing_one(group):
        missing_counts = group.isna().sum(axis=1)
        min_missing = missing_counts.min()
        candidates = group[missing_counts == min_missing]
        # if more than one, pick the first (or the row with smallest DATEDIFF)
        return candidates.iloc[[0]]  # keeps only 1 row

    print("Grouping rows by T1_PATH, finding the rows with the fewest missing values. If it's a tie between two rows, the first one for each T1_PATH group is kept")
    Tsub = Tsub.groupby('T1_PATH', group_keys=False).apply(fewest_missing_one, include_groups = True)
    Tsub.reset_index(drop=True, inplace=True)

    # Final sanity check
    dup_cols = [c for c in ['T1_PATH', 'MWC1T1_PATH'] if c in Tsub.columns]
    print(f"Final row count: {len(Tsub)}")
    print(f"Remaining duplicates: {Tsub.duplicated(subset=dup_cols).sum()}")
    
    # For each T1_PATH, see how many rows were kept
    rows_per_scan = Tsub.groupby('T1_PATH').size()
    print(rows_per_scan.value_counts())

    #Make the DATEDIFFS column numeric? Just because as datetime it says 'days' at the end and I don't like that.
    Tsub[f'DATEDIFFS_{column_suffix}'] = (Tsub[f'DATEDIFFS_{column_suffix}'].dt.days)
    
    print(Tsub[f'DATEDIFFS_{column_suffix}'].describe())
    print('')
    Tsub = Tsub.sort_values('T1_PATH').reset_index(drop=True)

    Tsub.to_csv('/rds/general/project/c3nl_scott_students/ephemeral/sankeith/scott_10k_scripts/alliedhealth/withamyloid_gregbased_scott10k_alliedhealth.csv', index = False)
    df = Tsub

Tsub.to_csv('/rds/general/project/c3nl_scott_students/ephemeral/sankeith/scott_10k_scripts/alliedhealth/debug_scott10k_alliedhealth.csv', index=False)



Length of the freshly joined dataframe: 23927
After dropping rows without T1 paths: 23927
Grouping rows by 'T1_PATH'. Within groups, organising rows so smallest DATE_DIFFS_AMYLOID are at the top
Finding rows where *EITHER* DATE_DIFFS_AMYLOID > 28 days *OR* there is no EXAMDATE_AMYLOID column. If that's the case, replace the relevant values with NaNs, NaTs and pd.NAs
Length of database before dropping exactly identical rows = 20903. Tsub is of type <class 'pandas.core.frame.DataFrame'>
Length of database after dropping rows which are exactly identical = 15733.
There are 0 rows with identical T1 MRI file paths, from 0 different MRI scans
Grouping rows by T1_PATH, finding the rows with the fewest missing values. If it's a tie between two rows, the first one for each T1_PATH group is kept
Final row count: 15733
Remaining duplicates: 0
1    15733
Name: count, dtype: int64
count    592.000000
mean      10.265203
std        8.745641
min        0.000000
25%        2.000000
50%        8.000000


### Checking diagnoses of scans in Scott 10K

In [20]:
import pandas as pd
df = pd.read_csv('/rds/general/project/c3nl_scott_students/ephemeral/sankeith/scott_10k_scripts/alliedhealth/debug_scott10k_alliedhealth.csv', low_memory = False)

print(df['DIAGNOSIS'].value_counts())


DIAGNOSIS
2.0    5057
1.0    4039
3.0    2836
Name: count, dtype: int64


In [22]:
%%bash

rsync -av /rds/general/project/c3nl_scott_students/ephemeral/sankeith/scott_10k_scripts/alliedhealth/debug_scott10k_alliedhealth.csv /rds/general/project/c3nl_scott_students/live/sankeith/scott_10k_housekeeping/no_neuroimaging_data_scott10k_alliedhealth.csv
cd /rds/general/project/c3nl_scott_students/live/sankeith/scott_10k_housekeeping
git add no_neuroimaging_data_scott10k_alliedhealth.csv
dategcp="`date +%d%b%Y`"
datetime="`date +%H%M%S`"
message="Rebuilt Scott 10K - from scratch to ADNI data (minus neuroimaging data): ${dategcp}, ${datetime}"

git commit -a -m "$message"
git push -u origin main

sending incremental file list

sent 93 bytes  received 12 bytes  210.00 bytes/sec
total size is 10,000,051  speedup is 95,238.58
[main a84603a] Rebuilt Scott 10K - from scratch to ADNI data (minus neuroimaging data): 19Feb2026, 200826
 Committer: Sankeith Kirubakaran <sk4724@cx3-2-11.cx3.hpc.ic.ac.uk>
s were configured automatically based
rate.ur username and hostname. Please check that they are accu
You can suppress this message by setting them explicitly:

Your Name"nfig --global user.name "
    git config --global user.email you@example.com

ntity used for this commit with:e ide

    git commit --amend --reset-author

tions(+), 9 deletions(-)ser
g_data_scott10k_alliedhealth.csv} (100%)_alliedhealth.csv => no_neuroimagin


To github.com:sankeithk/deconstructalz.github.io.git
   90799b3..a84603a  main -> main


branch 'main' set up to track 'origin/main'.
